awx/main/tasks.py

```diff
@@ -782,8 +790,10 @@ def add_ansible_venv(self, venv_path, env, add_awx_lib=True, **kwargs):
                 'a valid Python virtualenv does not exist at {}'.format(venv_path)
             )
         env.pop('PYTHONPATH', None)  # default to none if no python_ver matches
-        if os.path.isdir(os.path.join(venv_libdir, "python2.7")):
-            env['PYTHONPATH'] = os.path.join(venv_libdir, "python2.7", "site-packages") + ":"
+        for version in os.listdir(venv_libdir):
+            if fnmatch.fnmatch(version, 'python[23].*'):
+                if os.path.isdir(os.path.join(venv_libdir, version)):
+                    env['PYTHONPATH'] = os.path.join(venv_libdir, version, "site-packages") + ":"
```

Comment:
maybe some kind of `break` or a warning that it doesn't know exactly what path to use would be good in this case?

Reply:
What about an `else:` that logged a error telling you it couldn't find any `site-packages`.

Reply:
Granted, if you get this far, something is **very** wrong with your setup.

Reply:
> What about an else: that logged a error telling you it couldn't find any site-packages.

👍 👍 👍

Reply:
wait, let me amend that - an else: that logged an error addresses the case of not finding any packages.

There's also the error that they have >1 python libs available, like python2.7 and python3.6 both existing at the same time. We wouldn't know which one to use. I think we would use the latter, but knowing this would be good.

Reply:
Okay, I could see a `break` on the first match.  This example is a bit contrived, because if you have a `virtualenv` w/ this structure, you've done something _wrong_, but I think bailing out on the first match is a reasonable, deterministic behavior.